ros2bag/ros2bag/verb/record.py

```diff
@@ -168,8 +168,8 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
 
     def main(self, *, args):  # noqa: D102
         # both all and topics cannot be true
-        if (args.all and (args.topics or args.regex)) or (args.topics and args.regex):
-            return print_error('Must specify only one option out of topics, --regex or --all')
+        if (args.all and args.topics):
+            return print_error('Specify either --all or topics, but not both simultaneously.')
```

Comment:
There is an existing test case to check this

https://github.com/ros2/rosbag2/blob/ed8d98449320fd45ed3b7b78dbc584acfaae8d00/rosbag2_tests/test/rosbag2_tests/test_rosbag2_record_end_to_end.cpp#L665-L673